RELEASE_PROCEDURE.md

```diff
@@ -0,0 +1,39 @@
+# Release procedure
+
+## Preparation
+
+- Consolidate the [release notes](docs/root/version_history.md).
+  - Make sure breaking changes are explicitly called out.
+    Ideally these have been tracked on the go but it does not hurt to ask around and double
+    check. Running the python integration tests of the previous release with the latest binaries could serve as a double check for this.
+  - Ensure the release notes are complete. Ideally these have been tracked on the go.
+- Based on our [semantic versioning](https://semver.org/spec/v2.0.0.html) strategy, our release numbers look like `MAJOR.MINOR.PATCH`. Determine the release type to come up with what the next version should look like:
+  - Breaking changes increment `MAJOR`, and reset `MINOR.PATCH` to `0.0`
+  - Added functionality which is backwards compatible increments `MINOR`, and resets `PATCH` to `0`
+  - Bug fixes increment `PATCH`
+
+## Release steps
+
+1. Speculatively bump the version in [version.h](include/nighthawk/common/version.h) to the version you determined earlier. This may result in version gaps if a release attempt fails, but avoids having to freeze merges to master and/or having to work with release branches. In short it helps keeping the release procedure lean and mean and eliminates the need for blocking others while this procedure is in-flight.
+2. Draft a [GitHub tagged release](https://github.com/envoyproxy/nighthawk/releases/new). Earlier releases are tagged like `v0.1`, but as of `0.3.0`we are using [semantic versioning](https://semver.org/spec/v2.0.0.html)
+3. Perform thorough testing of the targeted revision to double down on stability [1]
+4. Create an optimized build for comparing with the previous release. Changes in performance
+  and/or measurement accuracy may need to be considered breaking. If in doubt, discuss!
+5. If things look good, [File an issue](https://github.com/envoyproxy/nighthawk/issues/new?title=[VOTE]+Release+v0.x&body=Release%20X%20is%20ready%20for%20review.%20Please%20take%20a%20look%20and%20vote!) to raise a vote on publishing the drafted release, and point contributors to that.
```

Comment:
(nit) Shall we update the link to include a semantic version number (e.g. v0.X.X).

Reply:
Done in 696410260a3dcfab4d9b48823fcfcff6ddc37b00